Source/GSXib5KeyedUnarchiver.m

```diff
@@ -0,0 +1,3391 @@
+ License as published by the Free Software Foundation; either
+ version 2 of the License, or (at your option) any later version.
+
+ This library is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.	 See the GNU
+ Lesser General Public License for more details.
+
+ You should have received a copy of the GNU Lesser General Public
+ License along with this library; see the file COPYING.LIB.
+ If not, see <http://www.gnu.org/licenses/> or write to the
+ Free Software Foundation, 51 Franklin Street, Fifth Floor,
+ Boston, MA 02110-1301, USA.
+ */
+
+#import "GSXib5KeyedUnarchiver.h"
+#import "GNUstepGUI/GSNibLoading.h"
+#import "GNUstepGUI/GSXibLoading.h"
+#import "GNUstepGUI/GSXibElement.h"
+
+#import "AppKit/NSApplication.h"
+#import "AppKit/NSBox.h"
+#import "AppKit/NSBrowser.h"
+#import "AppKit/NSBrowserCell.h"
+#import "AppKit/NSButtonCell.h"
+#import "AppKit/NSCell.h"
+#import "AppKit/NSClipView.h"
+#import "AppKit/NSFormCell.h"
+#import "AppKit/NSImage.h"
+#import "AppKit/NSMatrix.h"
+#import "AppKit/NSMenu.h"
+#import "AppKit/NSMenuItem.h"
+#import "AppKit/NSNib.h"
+#import "AppKit/NSParagraphStyle.h"
+#import "AppKit/NSPopUpButton.h"
+#import "AppKit/NSPopUpButtonCell.h"
+#import "AppKit/NSScroller.h"
+#import "AppKit/NSScrollView.h"
+#import "AppKit/NSSliderCell.h"
+#import "AppKit/NSSplitView.h"
+#import "AppKit/NSTableColumn.h"
+#import "AppKit/NSTableHeaderView.h"
+#import "AppKit/NSTableView.h"
+#import "AppKit/NSTabView.h"
+#import "AppKit/NSToolbarItem.h"
+#import "AppKit/NSView.h"
+#import "GSCodingFlags.h"
+
+//#define DEBUG_XIB5
+
+@interface NSCustomObject5 : NSCustomObject
+{
+  NSString *_userLabel;
+}
+
+- (NSString*) userLabel;
+@end
+
+@implementation NSCustomObject5
+
+static NSString *ApplicationClass = nil;
+
+- (id) initWithCoder: (NSCoder *)coder
+{
+  self = [super initWithCoder: coder];
+
+  if (self)
+    {
+      _userLabel = [coder decodeObjectForKey: @"userLabel"];
+
+      if (_className)
+        {
+          // If we've not set the general application class yet...
+          if (([NSClassFromString(_className) isKindOfClass: [NSApplication class]]) &&
+              (ApplicationClass == nil))
+            {
+              @synchronized([self class])
+                {
+                  ASSIGNCOPY(ApplicationClass, _className);
+                }
+            }
+        }
+
+      // Override thie one type...
+      if (_userLabel)
+        {
+          if ([@"Application" isEqualToString:_userLabel])
+            {
+              if (ApplicationClass == nil)
+                ASSIGN(_className, @"NSApplication");
+              else
+                ASSIGN(_className, ApplicationClass);
+            }
+        }
+    }
+
+  return self;
+}
+
+- (NSString *) userLabel
+{
+  return _userLabel;
+}
+
+@end
+
+@interface NSWindowTemplate5 : NSWindowTemplate
+{
+  BOOL _visibleAtLaunch;
```

Comment:
Can't `NSWindowTemplate` contain these two additional values? Based on the comment, I see why `IBActionConnection5` would be necessary, but have the value names changed for `NSWindowTemplate` too? Or are these just new values?

Or are we attempting to avoid to _serialize_ these into NIBs and pre-XIB5 files? I don't see serialization code, so that's hard to judge.

Reply:
I see you point and I tried to remove some of the other classes where similar changes where made but in this case I am unsure what is the bast way. I just didn't want to break old code.

Reply:
As long as the decision is explicitly made, this is fine.

It feels to me like we don't need an additional class here, but eh.

Reply:
OK, I will try to remove more of these in the future but for now we leave them in.